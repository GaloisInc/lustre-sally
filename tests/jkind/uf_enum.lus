-- This file exercises enumeration inputs to functions
-- Expected output:
-- VALID PROPERTIES: [ok]
-- INVALID PROPERTIES: [cex] (k = 1)

type color = enum { RED, GREEN, BLUE };

function f(c : color) returns (x : int);

node main(in : color) returns (ok, cex: bool);
let
  cex = not (f(RED) > f(GREEN) and f(GREEN) > f(BLUE));
  --%PROPERTY cex;

  ok = (f(in) = f(RED) or f(in) = f(GREEN) or f(in) = f(BLUE));
  --%PROPERTY ok;
tel;
