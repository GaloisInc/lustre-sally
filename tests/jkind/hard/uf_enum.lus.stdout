[[1;91mError[0m] Parse error at 10:6

